src/utils/sockctl.c

```diff
@@ -463,13 +463,16 @@ int writeread_domain_data_str(char *socket_path, const char *write_str,
   // rtrim modifies the input string.
   (void)rtrim(rec_data, NULL);
 
-  *reply = os_strdup(rec_data);
-  if (*reply == NULL) {
-    log_errno("os_strdup failed to copy string %s", rec_data);
+  char *trimmed_data = os_realloc(rec_data, strlen(rec_data) + 1);
+  if (trimmed_data == NULL) {
+    log_errno("os_realloc failed to rello string %s", rec_data);
```

Comment:
```suggestion
log_errno("os_realloc failed to realloc string %s", rec_data);
```

Reply:
Good catch! I'll use the suggestion below, since that has the correct indentation.

```suggestion
    log_errno("os_realloc failed to relloc string %s", rec_data);
```